packages/core/src/lib/context/createContext.ts

```diff
@@ -22,18 +22,26 @@ export function makeCreateContext({
   gqlNamesByList,
   config,
   lists,
-  cloudAssetsAPI,
 }: {
   graphQLSchema: GraphQLSchema;
   sudoGraphQLSchema: GraphQLSchema;
   config: KeystoneConfig;
   prismaClient: PrismaClient;
   gqlNamesByList: Record<string, GqlNames>;
   lists: Record<string, InitialisedList>;
-  cloudAssetsAPI: () => CloudAssetsAPI;
 }) {
-  const images = createImagesContext(config, cloudAssetsAPI);
-  const files = createFilesContext(config, cloudAssetsAPI);
+  const s3AssetsAPI =
+    config.storage && Object.values(config.storage).find(({ kind }) => kind === 's3')
+      ? s3Assets(config.storage)
+      : undefined;
```

Comment:
Could we not avoid doing things conditionally based on having 1 or more s3 assets vs 0?
```suggestion
  const s3AssetsAPI = s3Assets(config.storage);
```